### ISSUE TITLE
Extra table rows for `Returns` when empty lines start with spaces

### ISSUE THREAD

**Issue by LukeSavefrogs** [mkdocstrings/griffe#129]
## Describe the bug
When there are lines in the docstring that are **not completely empty** (but have spaces or tabs in it) between the end of the `Returns` block and the next block (in my case `Example`) **extra table rows are added**.

## To Reproduce
1. Write the following in a docstring: 
    ```python
    def test():
        """Test docstring with 2 extra empty parameters.
    
        Returns:
            str: TEST
            
            
        Example:
            ```pycon
            >>> test()
            ```
        """
    ```
2. A similar output should be shown:
    ![image](https://user-images.githubusercontent.com/33452387/218729548-12d4b17f-a2ef-4746-a75b-1e7957f39305.png)

This however does not happen if there are no spaces (only empty lines) between the last parameter and the next block (in this case `Example`)
```python
def test():
    """Test docstring with 2 extra empty parameters.

    Returns:
        str: TEST


    Example:
        ```pycon
        >>> test()
        ```
    """
```
![image](https://user-images.githubusercontent.com/33452387/218730557-f9649b1d-686e-4181-9d06-53b935a1cd88.png)



## System information
- `griffe` version: `0.25.4`
- Python version: `3.9`
- OS: Windows

**Comment by LukeSavefrogs** [comment]
I think this may be related to something going on in these lines:
https://github.com/mkdocstrings/griffe/blob/1ff091890f4ebdf11709326028ef9bc74f1f1455/src/griffe/docstrings/google.py#L667-L682

**Comment by LukeSavefrogs** [comment]
# Solution (?)
I made some progress...

I found that the filtering on `docstring.lines` did not work because the function `_read_returns_section` (which is responsible for the creation of the returns section) read directly from the original `docstring`.

I solved temporarily by changing this line:
https://github.com/mkdocstrings/griffe/blob/1ff091890f4ebdf11709326028ef9bc74f1f1455/src/griffe/docstrings/google.py#L393

To:
```python
if name is not None:
    returns.append(DocstringReturn(name=name or "", annotation=annotation, description=description))

```

This way the bug **seems fixed**:
![image](https://user-images.githubusercontent.com/33452387/218778683-77c4f244-1d72-4bbb-8d0a-6c355c8320fa.png)

# Considerations
- It would be better to move the check at the start, instead of right before the assignment (maybe something like `if _is_empty_line(return_lines[0]): continue`)
- Should the other `_read_*section` functions be "patched" as well?

**Comment by pawamoy** [comment]
Hello, thanks for the report and investigation!

I've looked at it and the bug actually comes from the function that splits blocks of text (sections). If you look at this:

https://github.com/mkdocstrings/griffe/blob/1ff091890f4ebdf11709326028ef9bc74f1f1455/src/griffe/docstrings/google.py#L112-L119

...you might notice that the first condition does not check if the line is "empty" (just whitespace). So the solution would simply be to check that, and update the rest of the checks accordingly.

I should be able to push a fix tonight :slightly_smiling_face:

**Comment by LukeSavefrogs** [comment]
Thank you @pawamoy for the quick fix!

Do you know when will this fix be available in a release?

**Comment by pawamoy** [comment]
Just released 0.25.5 :slightly_smiling_face:
